File: repositories/note_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type NoteRepository interface {
	Create(ctx context.Context, note *entities.DoctorNote) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.DoctorNote, error)
}

type GormNoteRepository struct {
	db *gorm.DB
}

func NewNoteRepository(db *gorm.DB) NoteRepository {
	return &GormNoteRepository{db: db}
}

func (r *GormNoteRepository) Create(ctx context.Context, note *entities.DoctorNote) error {
	return r.db.WithContext(ctx).Create(note).Error
}

func (r *GormNoteRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.DoctorNote, error) {
	var notes []entities.DoctorNote
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Order("created_at DESC").Find(&notes).Error
	return notes, err
}

File: python_backend/repositories/note_repository.py
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.note import DoctorNote

class NoteRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, note: DoctorNote) -> None:
        self.db.add(note)
        self.db.commit()
        self.db.refresh(note)

    def FindByDoctorID(self, doctor_id: UUID) -> List[DoctorNote]:
        return self.db.query(DoctorNote).filter(DoctorNote.doctor_id == doctor_id).order_by(DoctorNote.created_at.desc()).all()
